### PR TITLE
Attempt to convert imported signer Keystore to a JKS.

### PR DESCRIPTION
**Pull request by renatoathaydes**
Signicat will attempt to use it as JKS, so this conversion must be performed explicitly.

### CODE REVIEW

**Review by daniellindau — APPROVED** [review]
Looks good!

File: src/main/kotlin/io/curity/identityserver/plugin/signicat/authentication/util.kt [hunk 1]
```diff
@@ -35,4 +38,34 @@ internal fun withEnvironment(config: SignicatAuthenticatorPluginConfig): String
...
+ * Signicat expects to be able to load the signre truststore as a JKS keystore specifically, so this function must be
+ * called to make sure the [Keystore] provided to [com.signicat.services.client.saml.SamlFacade#setSamlKeystore] does
+ * not cause any errors.
+ *
+ * Only the original keystore's certificate is added to the return [KeyStore]'s byte array.
+ */
+internal fun SignerTrustCryptoStore.convertToJksByteArray(): ByteArray
+{
+    val jks = convertToJks()
+    val stream = ByteArrayOutputStream()
```
Comment (travisspencer) [inline src/main/kotlin/io/curity/identityserver/plugin/signicat/authentication/util.kt:53]: this doesn't need to be closed?

File: src/main/kotlin/io/curity/identityserver/plugin/signicat/authentication/SignicatCallbackRequestHandler.kt [hunk 2]
```diff
@@ -196,30 +196,22 @@ class SignicatCallbackRequestHandler(config : SignicatAuthenticatorPluginConfig)
...
-            val stream = ByteArrayOutputStream()
-
-            ObjectOutputStream(stream).use { out ->
-                out.writeObject(store.asKeyStore)
-            }
-
-            val bytes = stream.toByteArray()
-            
-            samlFacade.setSamlKeystore(bytes)
+            samlFacade.setSamlKeystore(store.convertToJksByteArray())
```
Comment (travisspencer) [inline src/main/kotlin/io/curity/identityserver/plugin/signicat/authentication/SignicatCallbackRequestHandler.kt:214]: what if the keystore is already a `jks` keystore? Then we do this work for nothing.

**Review by travisspencer — APPROVED** [review]

File: src/main/kotlin/io/curity/identityserver/plugin/signicat/authentication/util.kt (hunk 1 above)
Comment (renatoathaydes) [reply src/main/kotlin/io/curity/identityserver/plugin/signicat/authentication/util.kt:53]: No, it's just an in-memory stream.

File: src/main/kotlin/io/curity/identityserver/plugin/signicat/authentication/SignicatCallbackRequestHandler.kt (hunk 2 above)
Comment (renatoathaydes) [reply src/main/kotlin/io/curity/identityserver/plugin/signicat/authentication/SignicatCallbackRequestHandler.kt:214]: I check for that now and avoid doing any work in that case.

**Pull request merged**